addons/account/tests/test_multivat.py

```diff
@@ -0,0 +1,158 @@
+from odoo.tests import tagged
+from odoo.tests.common import TransactionCase
+
+
+def _get_chart_template_mapping(self, get_all=False):
+    return {
+        'local': {
+            'name': 'local',
+            'country_id': self.env.ref("base.be").id,
+            'country_code': 'be',
+            'modules': ['account'],
+            'parent': None,
+        },
+        'foreign': {
+        'name': 'foreign',
+            'country_id': self.env.ref("base.fr").id,
+            'country_code': 'fr',
+            'modules': ['account'],
+            'parent': None,
+        }
+    }
+
+def data_method_provider(chart_template_name, country_code):
+    country = f"base.{country_code}"
+    external_id_prefix = '' if chart_template_name == 'local' else f"{chart_template_name}_"
+
+    def test_get_data_getter(self, template_code):
+        return {
+            'template_data': {
+                'code_digits': 6,
+                'currency_id': 'base.EUR',
+                'property_account_income_categ_id': f'{external_id_prefix}test_account_income_template',
+                'property_account_expense_categ_id': f'{external_id_prefix}test_account_expense_template',
+            },
+            'res.company': {
+                self.env.company.id: {
+                    'bank_account_code_prefix': '1000',
+                    'cash_account_code_prefix': '2000',
+                    'transfer_account_code_prefix': '3000',
+                },
+            },
+            'account.account': {
+                f'{external_id_prefix}test_account_income_template': {
+                    'name': f'{external_id_prefix} income',
+                    'code': '111110',
+                    'account_type': 'income',
+                },
+                f'{external_id_prefix}test_account_expense_template': {
+                    'name': f'{external_id_prefix} expense',
+                    'code': '222221',
+                    'account_type': 'expense',
+                },
+            },
+            'account.journal': self._get_account_journal(template_code),
+            'account.tax.group': {
+                f'{external_id_prefix}tax_group_taxes': {
+                    'name': f"{external_id_prefix}Taxes",
+                    'sequence': 0,
+                    'country_id': country,
+                },
+            },
+            'account.tax': {
+                xmlid: _tax_vals(name, amount, external_id_prefix)
+                for name, xmlid, amount in [
+                    (f'{external_id_prefix}Tax 1', f'test_tax_1_template', 15),
+                    (f'{external_id_prefix}Tax 2', f'test_tax_2_template', 0),
+                ]
+            },
+        }
+    return test_get_data_getter
+
+def _tax_vals(name, amount, external_id_prefix):
+    return {
+        'name': name,
+        'amount': amount,
+        'type_tax_use': 'purchase',
+        'repartition_line_ids': [
```

Comment:
it's boring that this need to be done.
Maybe check with WAN if the declaration could be done without it someway or another